doc/nrf/gs_modifying.rst

```diff
@@ -95,15 +122,80 @@ Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c`
 Configuring your application
 ****************************
 
-If your application uses a provided library or targets a specific board, you might want to change the default configuration of the library or board.
+You might want to change the default options of the application.
 There are different ways of doing this, but not all will store your configuration permanently.
 
-The default configuration for a library is specified in its :file:`Kconfig` file.
-Similarly, the default configuration for a board is specified in its :file:`*_defconfig` file (and its :file:`Kconfig.defconfig` file, see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
-The configuration for your application, which might override some default options of the libraries or the board, is specified in a :file:`prj.conf` file in the application directory.
+Zephyr's configuration system is distributed, and so is the |NCS| one.
+This means that there are multiple sources of configuration.
+In the |NCS|, the configuration system consists of the following elements:
+
+* Kconfig configuration files that focus on software-related options.
+* Devicetree specification (DTS) configuration files that focus on hardware-related options.
+* Partition manager configuration files that focus on memory layout.
+
+Software-related configuration
+==============================
+
+The software-related configuration comes from Kconfig files (called simply :file:`Kconfig`).
+There are several layers that provide the configuration options:
+
+* The lowest layer consists of defaults provided in Kconfig files of all system components, libraries, and application itself.
+* The defaults are overlayed by the options specified by the selected board in its :file:`*_defconfig` file and its :file:`Kconfig.defconfig` file (see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
+* The configuration is additionally overlayed by the :file:`prj.conf` file provided by the application.
+
```

Comment:
I'm not sure this helps the reader much.

Maybe we should describe the two parts individually.

- The `Kconfig` and `Kconfig.*` files in the build system forms a Kconfig tree describing the system.
  It also controls the default value for settings not configured by the application or the user.

- The board specific `*_defconfig` and applications `prj.conf` contains configuration settings specific to application being build.

@SebastianBoe what do you think, wouldn't it be better to split this into those two parts ?

Reply:
https://github.com/nrfconnect/sdk-nrf/pull/6121#pullrequestreview-818914020